src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -388,40 +364,38 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
         }
 
         public async Task SwitchToAvailablePackagesAsync() {
-            await _coreShell.SwitchToMainThreadAsync();
-            if (_selectedTab == SelectedTab.AvailablePackages) {
-                return;
-            }
-
-            _selectedTab = SelectedTab.AvailablePackages;
-            IsLoading = !_availableLock.IsCompleted;
-
-            await EnsureAvailablePackagesLoadedAsync();
-            if (_selectedTab == SelectedTab.AvailablePackages) {
-                IsLoading = false;
-                ReplaceItems(_availablePackages);
+            if (await SetTabAsync(SelectedTab.AvailablePackages)) {
+                if (!_availableLock.IsCompleted) {
+                    await EnsureAvailablePackagesLoadedAsync();
+                }
+                await ReplaceItemsAsync();
             }
         }
 
-        private async Task EnsureAvailablePackagesLoadedAsync() {
-            await _coreShell.SwitchToMainThreadAsync();
-
+        private async Task EnsureAvailablePackagesLoadedAsync(bool updateStateOnly = false) {
             var availablePackagesLoaded = await _availableLock.WaitAsync();
-            if (!availablePackagesLoaded) {
-                try {
-                    await LoadAvailablePackagesAsync();
-                } catch (RPackageManagerException ex) {
-                    _coreShell.DispatchOnUIThread(() => AddErrorMessage(ex.Message));
-                } finally {
-                    _availableLock.Release();
+            try {
+                if (!availablePackagesLoaded || updateStateOnly) {
+                    await LoadAvailablePackagesAsync(updateStateOnly);
                 }
+            } catch (RPackageManagerException ex) {
+                _coreShell.DispatchOnUIThread(() => AddErrorMessage(ex.Message));
+            } finally {
+                _availableLock.Release();
             }
         }
 
-        private async Task LoadAvailablePackagesAsync() {
+        private async Task LoadAvailablePackagesAsync(bool updateStateOnly = false) {
```

Comment:
Not sure I understand this. If available packages are loaded,  state of every package will be updated in `ReloadInstalledAndLoadedPackagesAsync`, so if `updateStateOnly == true`, there is no reason to call this method at all